src/Compiler.php

```diff
@@ -240,6 +241,19 @@ private function getParameter($name, $type, &$output)
         return true;
     }
 
+    private function getSubtractiveParameters($nameA, $nameB, $typeA, $typeB, &$outputA, &$outputB)
+    {
+        list($idA, $idB) = $this->arguments->useSubtractiveArgument($nameA, $nameB, $typeA, $typeB);
+
+        if ($idA === null || $idB === null) {
```

Comment:
You can improve readability (and rarely prevent bugs) by making the order of evaluation explicit, like this:
```php
($idA === null) || ($id === null)
```

But, in this case, there is built-in function:
```php
if (!isset($idA, $idB)) {
   ...
}
```